node/lib/util/git_util.js

```diff
@@ -911,3 +911,66 @@ exports.configIsTrue = co.wrap(function*(repo, configVar) {
     return (configured === "true" || configured === "yes" ||
             configured === "on");
 });
+
+/**
+ * Fetch the remote having the specified `repo`.
+ * Throw a `UserError` object if the repository cannot be fetched.
+ *
+ * @async
+ * @param {NodeGit.Repository} repo
+ */
+exports.getRefs = co.wrap(function *(repo) {
+	assert.instanceOf(repo, NodeGit.Repository);
+
+    const execString = `git -C '${repo.path()}' ls-remote --refs`;
```

Comment:
"... -- refs/" instead "... -- refs" (that is, add a trailing slash)

Reply:
`"... -- refs/" instead "... -- refs" (that is, add a trailing slash)`

Not sure what you mean by this, why  -- refs /? ls-remote takes --refs flag.

Reply:
Oops, I totally misread this.  Nevermind.